libs/platform/src/lib/components/menu/menu-item.component.scss

```diff
@@ -1,131 +1,31 @@
-    align-items: center;
-    margin: 6px 4px 6px 8px;
-    color: var(--sapInformationColor, $fd-icon-color);
-    background-color: transparent; //recent styles repo changes made it white color
-}
-
-.fd-menu--addon-before > .fd-menu__item {
-    //fd-menu--addon-before represents the container of the menu item
-    padding-left: 0px;
-}
-
-.fd-menu__addon-after {
-    //fd-menu__addon-after represents the add-on icon element at the end of the item
-    width: 28px;
-    height: 28px;
-    padding: 8px; //center-aligned icon, therefore 8px
-    margin: 6px 4px 6px 8px;
-    color: var(--sapInformationColor, $fd-icon-color);
-    display: flex;
-    float: right;
-    clear: right;
-    justify-content: center;
-    align-items: center;
-}
-
-[dir='rtl'] .fd-menu__addon-after {
-    float: left;
-    clear: left;
-}
-.is-disabled,
-.is-disabled:hover,
-.is-disabled:focus,
-.is-disabled:active {
-    cursor: not-allowed;
-    text-decoration: none;
-    pointer-events: none;
-    .fd-menu__item,
-    .fd-menu__addon-before,
-    .fd-menu__addon-after {
-        color: var(--sapContent_DisabledTextColor, $fd-disabled-item-color);
-    }
-}
-
-//inline-flex cannot be used in conjunction with overflow:hidden. This is a workaround to have both.
-.fd-menu__label-truncation {
-    display: inline-flex;
-}
-
-//have the right state for selected item
-.selected-item-state {
-    background-color: var(--sapInformationColor, $fd-selected-item-bg-color);
-    &:hover {
-        .fd-menu__item {
-            font-weight: bold;
-            color: $fd-item-color;
-        }
-        background-color: var(--sapInformationColor, $fd-selected-item-bg-color-hover);
-    }
-    &:focus {
-        outline: var(--sapButton_Emphasized_Hover_BorderColor, $fd-item-focus-color) dotted 1px;
-        .fd-menu__item {
-            color: $fd-item-color; //to override the blue color to selected item
-        }
-    }
-}
-
-//have the right state for normal/unselected items
-.normal-item-state {
-    &:hover {
-        background-color: var(--sapList_FooterBackground, $fd-normal-item-color-hover);
-        .fd-menu__item {
-            font-weight: bold;
-            color: $fd-item-color;
-        }
-    }
-    &:focus {
-        outline: var(--sapButton_Emphasized_Hover_BorderColor, $fd-item-focus-color) dotted 1px;
-        .fd-menu__item {
-            color: $fd-item-color; //to override the blue color to selected item
-        }
-    }
-}
-
-//removes extra padding when icons are present
-.menu-icon-before + .fd-menu__label-truncation {
-    .fd-menu__item {
-        // padding: 12px 0px;   //for alignment with icons
-        padding: 8px 0px;
-    }
+:host.cascades-right {
+  padding-right: 2rem;
+  /* render arrow for trigger */
+  &.trigger:after {
+    position: absolute;
+    right: 0;
+    width: 2rem;
+    font-family: SAP-icons;
+    font-style: normal;
+    font-weight: 400;
+    font-size: 0.75rem;
+    text-align: center;
+    content: '\e066';
```

Comment:
Fiori team says although right arrow should be used to show a submenu, we shouldn't be restricting it to this particular icon in our code.  @droshev could you please confirm if using it in this way is okay or should we provide developer the means to place his own icon instead?

Reply:
> 1. When navigating through mouse, on clicking a submenu item, the parent menu closes. But when navigating through keyboard, on clicking a submenu item, the parent menu does not close. Shouldn't this behaviour be consistent?

All parent menus should close when "mouse clicking" or  "keyboard typing enter" on a leaf node.
I can't seem to reproduce the behavior you're describing.

> 2. Can we add an example that shows icons, separator lines and a longer menu with scrolling inside the menu; basically something like what the earlier 'complex menu' example had?

I'll try and add the additional examples. Scrolling has not been implemented - I'll try to add that feature as well.

Reply:
@kavya-b I think I'm going to wait for the fundamental-styles update (0.8.x? the one with updated Menu styles) before adding examples with icons and separator lines. I'm guessing I'll have to redo the component templates when the update occurs.